### PR TITLE
Run AOT tests against reference system

### PR DESCRIPTION
**Pull request by Mousius**
This introduces an alternative way of running AOT tests using the reference system added in https://github.com/apache/tvm/pull/8514. This gives us additional assurance that the AOT output runs successfully on embedded platforms in our core test suite.

I've also changed calculate_workspace_sizes to debug_workspace_sizes and default to False in most cases as it only needs to be True for a few cases to check theoutput with the debug flag - this was discovered trying to allocate 16MB in an embedded test :scream_cat:

Co-authored-by: Grant Watson <grant.watson@arm.com>

### CODE REVIEW

**Comment by areusch** [comment]
@Mousius just wondering if any of this is able to be handled by Zephyr, or if the FVP is outside Zephyr?

**Comment by Mousius** [comment]
Hi @areusch, we could use Zephyr to wrap this up but I think it makes sense in this case to match this closely to the existing AOT test utils rather than introducing the dependency on Zephyr and generating a Zephyr project to drive via `west`. The other issue is that the Zephyr infrastructure isn't present in `ci_cpu` where-as the files introduced alongside the FVP in https://github.com/apache/tvm/pull/8514 are.

**Comment by Mousius** [comment]
@areusch / @leandron I think this is pending a Docker update to add the files from https://github.com/apache/tvm/pull/8514 ?

File: tests/python/relay/aot/aot_test_utils.py [hunk 1]
```diff
@@ -64,6 +68,38 @@ class AOTTestModel(NamedTuple):
...
+    """
+
+    makefile: str = "default"
+    prologue: str = ""
+    includes: List[str] = []
+    parameters: Dict[str, str] = {}
+
+
+AOT_DEFAULT_RUNNER = AOTTestRunner()
+AOT_CORSTONE300_RUNNER = AOTTestRunner(
```
Comment (manupa-arm) [inline tests/python/relay/aot/aot_test_utils.py:93]: Maybe its worth linking here for more info about the reference system ?

**Review by manupa-arm — CHANGES_REQUESTED** [review]
Thanks @Mousius @grant-arm 

Mostly looks good! just one suggestion to link to information of reference system.

File: tests/python/relay/aot/aot_test_utils.py (hunk 1 above)
Comment (Mousius) [reply tests/python/relay/aot/aot_test_utils.py:96]: Great suggestion, I've added this :smile_cat:

**Review by manupa-arm — APPROVED** [review]
LGTM

**Review by jroesch — APPROVED** [review]

**Pull request merged**